cmd.go

```diff
@@ -137,59 +139,33 @@ func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
 	return r.prevPreRunE(cmd, args)
 }
 
-func (r *Runner) runE(runner interface{}, cfg interface{}) func(cmd *cobra.Command, args []string) error {
+func (r *Runner) runE(module fx.Option) func(cmd *cobra.Command, args []string) error {
 	return func(cmd *cobra.Command, args []string) error {
-		// Unmarshal the configuration
-		if err := Unmarshal(r.prefix, cfg); err != nil {
-			return errors.Wrap(err, "unable to unmarshal configuration")
-		}
-
 		l := log.New()
 
 		loglevelFlag, err := cmd.Flags().GetString("loglevel")
 		if err != nil {
 			return err
 		}
 
-		if module, ok := runner.(fx.Option); ok {
-			if _, ok = cfg.(config.Config); ok {
-				runner = func(ctx context.Context, cfg config.Config) error {
-					app := fx.New(
-						logging.WithLogger(l),
-						FxContext(ctx),
-						FxOptions(cfg),
-						fx.Supply(cmd),
-						fx.Supply(service.Name(r.prefix)),
-						fx.Supply(logging.Level(loglevelFlag)),
-						Module,
-						module,
-					)
-
-					app.Run()
-
-					return app.Err()
-				}
-			} else {
-				panic("if providing an fx Module, then config must implement ProvidesFxOptions")
-			}
-		}
-
-		// Call the runner
-		out := reflect.ValueOf(runner).Call([]reflect.Value{
-			reflect.ValueOf(log.ToContext(cmd.Context(), l)),
-			reflect.ValueOf(cfg),
-		})
-
-		// Handle any result
-		if len(out) > 0 && out[0].IsValid() && !out[0].IsNil() {
-			e := out[0].MethodByName("Error")
-			out = e.Call([]reflect.Value{})
-			if len(out) > 0 && out[0].IsValid() {
-				return errors.New(out[0].String())
-			}
-		}
-
-		return nil
+		app := fx.New(
+			logging.WithLogger(l),
+			FxContext(cmd.Context()),
+			fx.Supply(cmd),
+			fx.Supply(service.Name(r.prefix)),
+			fx.Supply(logging.Level(loglevelFlag)),
+			Module,
+			module,
+			fx.Invoke(
+				func(provider config.Provider) {
+					provider.Load(cmd.Context())
+				},
+			),
```

Comment:
Curious - If this is the correct way to call `Load`?

Reply:
No, inject the ctx as well, so it would be:
```
func(ctx context.Context, provider config.Provider) error {
	return provider.Load(ctx)
},
```
And actually this function could probably be moved to the config/module.go and you should return an error at least